src/vision/tenets.md

```diff
@@ -0,0 +1,20 @@
+# Design tenets for async
+
+The design tenets describe the key principles that drive our work on async. Hopefully, we are able to achieve and honor all of them all of the time. Sometimes, though, they come into conflict, and we have to pick -- in that case, we prefer the tenet earlier in the list.
+
+1. **Minimal overhead.** Rust Async I/O performance should compare favourably with any other language. In the extreme case, it should be possible to use async/await without any required allocation, although this is unlikely to be a common case in production systems.
+2. **Async is like sync, but with blocking points clearly identified.** At the highest level, writing a simple program using asynchronous I/O in Rust should be analogous to writing one that uses synchronous I/O, except that one adds `async` in front of function declarations and adds `.await` after each call. We should aim for analogous design between synchronous and asynchronous equivalents. Similarly, streams should be like asynchronous iterators. One should be able to use the same sort of combinators with streams and to iterate over them in analogous ways.
```

Comment:
+1. There are some forms of concurrent control flow that only exist in async and has no synchronous counterpart (such as join, select, futuresunordered). In other words I think we can achieve sync -> async being a mechanical refactor, but async -> sync cannot always be mechanical.